File: src/00/sigma_core/src/ontology_gen/00/dir8_x.rs

pub fn dir8_x(n: i32) -> i32 {
    if n == 0 || n == 4 || n == 6 {
        -1
    } else if n == 1 || n == 5 || n == 7 {
        1
    } else {
        0
    }
}

File: src/_as/00/dir8_x.ts
// SSoT: src/ontology/spatial/dir8_x.md

@inline
export function dir8_x(n: i32): i32 {
if (n == 0 || n == 4 || n == 6) return -1;
if (n == 1 || n == 5 || n == 7) return 1;
return 0;
}
